autotest/pyscripts/test_gdal_utils_cli.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env pytest
+# -*- coding: utf-8 -*-
+###############################################################################
+#
+# Project:  GDAL/OGR Test Suite
+# Purpose:  osgeo_utils (gdal-utils) testing
+# Author:   Matt Wilkie <maphew@gmail.ca>
+#
+###############################################################################
+# Copyright (c) 2022, Matt Wilkie <maphew@gmail.ca>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+'''
+*** STUB *** Test that command line gdal-utils are in PATH and run. We only
+test script starts and returns expected defaults. We do not test if script
+functions properly.
+
+Tested:
+    gdalcompare <no args>
+Returns:
+    Usage: gdalcompare.py [-sds] <golden_file> <new_file>
+
+Not tested:
+    gdalcompare file1.tif file2.tif
+Returns:
+    <valid results>
+
+Resources used:
+ - https://github.com/rhcarvalho/pytest-subprocess-example
+ - https://stackoverflow.com/questions/30528099/unittesting-python-code-which-uses-subprocess-popen
+ - https://www.digitalocean.com/community/tutorials/how-to-use-subprocess-to-run-external-programs-in-python-3
+ - https://stackoverflow.com/questions/13493288/python-cli-program-unit-testing
+'''
+import sys
+import subprocess
+import pytest
+
+# Skip if gdal-utils is not known to pip (and therefore not registered in
+# python 'site-packages' and 'Scripts')
+installed = subprocess.run([sys.executable, '-m', 'pip', 'show', 'gdal-utils'])
+if not installed.returncode == 0:
+    pytest.skip("The 'gdal-utils' package is not installed.", allow_module_level=True)
+
+
+@pytest.mark.parametrize("input,want", [
+    pytest.param("gdalcompare", {
+        "returncode": 1,
+        "stdout": "Usage: gdalcompare.py [-sds] <golden_file> <new_file>\n",
```

Comment:
relying on an exact string is going to be fragile when new options are tested. I would just check that stdout or stderr starts with 'Usage:'

Reply:
I like that idea. I've spent the last several sessions figuring out how to generate and then read stdout/err from files in pytest. I learned a lot! but still didn't get to the finish line (c.f. a93a1ed). I'm not sorry for the field trip, as it's stretched and extended my skills, but I do want to finish and take time look at the flowers later. ;-)

Reply:
hmmm. It's not going to be as straightforward as seeking for 'Usage'. Every program is different in what they return. Arguably they should be harmonized to all use a common syntax. It would also be good to have them settle on a single one of stdout or stderr instead of back and forth like they are now.

Reply:
An alternate approach would be checking that the stderr output does not contain "^Traceback" (or whatever else is common when exiting due to uncaught exceptions) and that something was printed to either stderr or stdout.

Reply:
Since the test using the external data files approach is working and done I've committed it. That can work until the usage messages are standardized, which I think is worthwhile on it's merit.

Better yet, this test script has proofed it's worth! `gdal_merge` is broken in this branch of _gdal-utils_.